MapboxCoreNavigation/RouteController.swift

```diff
@@ -277,10 +279,11 @@ open class RouteController: NSObject, Router {
         return false
     }
     
-    internal func userIsWithinRadiusOfRoute(location: CLLocation) -> Bool {
+    internal func userIsWithinEnvelope(of step: RouteStep, at location: CLLocation) -> Bool {
         let radius = max(reroutingTolerance, RouteControllerManeuverZoneRadius)
-        let isCloseToCurrentStep = location.isWithin(radius, of: routeProgress.currentLegProgress.currentStep)
-        return isCloseToCurrentStep
+        let isCloseToCurrentStep = location.isWithin(radius, of: step)
+        let isWithinCourseTolerances = userCourseIsOnRoute(location)
```

Comment:
Did we determine whether the course-related heuristics in `userCourseIsOnRoute(_:)` are germane to the issue we’re attempting to fix?